pallets/handles/src/lib.rs

```diff
@@ -140,6 +132,42 @@ pub mod pallet {
 	pub type CanonicalBaseHandleToSuffixIndex<T: Config> =
 		StorageMap<_, Blake2_128Concat, Handle, SequenceIndex, OptionQuery>;
 
+	/// PayloadSignatureRegistryList is used to prevent replay attacks for extrinsics
+	/// that take an externally-signed payload.
+	/// For this to work, the payload must include a mortality block number, which
+	/// is used in lieu of a monotonically increasing nonce.
+	///
+	/// The list is forwardly linked. (Example has a list size of 3)
+	/// - signature, forward pointer -> n = new signature
+	/// - 1,2 -> 2,3 (oldest)
+	/// - 2,3 -> 3,4
+	/// - 3,4 -> 4,5
+	/// -   5 -> n (newest in pointer data)
+	/// ### Storage
+	/// - Key: Signature
+	/// - Value: Tuple
+	///     - `BlockNumber` when the keyed signature can be ejected from the registry
+	///     - [`MultiSignature`] the forward linked list pointer. This pointer is the next "newest" value.
+	#[pallet::storage]
+	#[pallet::getter(fn get_payload_signature_registry)]
+	pub(super) type PayloadSignatureRegistryList<T: Config> = StorageMap<
+		_,                                // prefix
+		Twox64Concat,                     // hasher for key1
+		MultiSignature, // An externally-created Signature for an external payload, provided by an extrinsic
+		(T::BlockNumber, MultiSignature), // An actual flipping block number and the oldest signature at write time
+		OptionQuery,                      // The type for the query
+		GetDefault,                       // OnEmpty return type, defaults to None
+		T::MaxSignaturesStored,           // Maximum total signatures to store
+	>;
+
+	/// This is the pointer for the Payload Signature Registry
```

Comment:
A lot of these prevention measures are replica of what `msa` is doing

Reply:
Do we have an interface with `msa` for this ? Or a separate signature registry is needed for `handles`

Reply:
True, there is no need for registry

Reply:
I think it is an open question if it is worth forcing that users do not change their handle more than once per `max_mortality`. Thoughts?

Reply:
Thinking from perspective of usability and end use enagagement, having `max_mortality` is definitely justified, any number of blocks in mind?

Reply:
Likely just setup as the same in Config as we do with the signature registry for the msa pallet.

Changing this would also require that we store the block number on the msaToDisplayName right?

Reply:
Yep, that and benchmark updates is in coming commit